Update Dynamo passrate/histogram scripts

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #118752

Changelog:
- Don't count running PYTORCH_TEST_WITH_DYNAMO=1 on dynamo/ tests in the pass
rate. This was a bug (we were counting all of these as failing, but in
reality, most of these pass). The net effect is that the passrate is (artifically)
6% higher.
- Have the histogram script filter out skips based on the passrate metric.

zou3519 committed:
Update Dynamo passrate/histogram scripts

Changelog:
- Don't count running PYTORCH_TEST_WITH_DYNAMO=1 on dynamo/ tests in the pass
rate. This was a bug (we were counting all of these as failing, but in
reality, most of these pass). The net effect is that the passrate is (artifically)
6% higher.
- Have the histogram script filter out skips based on the passrate metric.

[ghstack-poisoned]

jamesjwu commented on scripts/compile_tests/failures_histogram.py:
```diff
@@ -73,21 +56,32 @@ def repro(testcase):
     )
-    parser.add_argument("directory")
+    # linux-focal-py3.11-clang10 (default) Test Reports (xml) directory
+    parser.add_argument("eager_dir")
```
It may be helpful to add `help=""` with the comment too so that failures_histogram.py --help works

jamesjwu commented on scripts/compile_tests/failures_histogram.py:
```diff
@@ -73,21 +56,32 @@ def repro(testcase):
-    parser.add_argument("directory")
+    # linux-focal-py3.11-clang10 (default) Test Reports (xml) directory
+    parser.add_argument("eager_dir")
+    # linux-focal-py3.8-clang10 (dynamo) Test Reports (xml) directory
```
Is there a reason this one is py3.8 but the one above is py3.11? Should we not use the same py version for both? 

Maybe the fully correct thing is to pass in both py3.11 and py3.8 for both, and then the union of all test failures is the comprehensive list. But that might be overkill.

zou3519 replied:
whoops, both should be py3.11. Yes, the union of both would be better. But we have pretty similar failures across py311 and py38 and downloading test reports is already a bit of a pain, so maybe in the future.

jamesjwu reviewed (approved):
I'm getting about a ~11% passrate based on https://gist.github.com/jamesjwu/cc80d7da305add0a69c5e39aae09a077. Is that the number you're expecting, or did I pass in the wrong inputs?

zou3519 committed:
Update on "Update Dynamo passrate/histogram scripts"


Changelog:
- Don't count running PYTORCH_TEST_WITH_DYNAMO=1 on dynamo/ tests in the pass
rate. This was a bug (we were counting all of these as failing, but in
reality, most of these pass). The net effect is that the passrate is (artifically)
6% higher.
- Have the histogram script filter out skips based on the passrate metric.

[ghstack-poisoned]

zou3519 committed:
Update on "Update Dynamo passrate/histogram scripts"


Changelog:
- Don't count running PYTORCH_TEST_WITH_DYNAMO=1 on dynamo/ tests in the pass
rate. This was a bug (we were counting all of these as failing, but in
reality, most of these pass). The net effect is that the passrate is (artifically)
6% higher.
- Have the histogram script filter out skips based on the passrate metric.

[ghstack-poisoned]

zou3519 commented:
@pytorchbot merge -f "passed lint; isn't tested in CI otherwise; merging to unblock next PR in stack"
